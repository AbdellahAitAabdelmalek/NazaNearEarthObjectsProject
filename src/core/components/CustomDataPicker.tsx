import React, {useState} from 'react';
import {View, Button,Text, Platform} from 'react-native';
import DateTimePicker from '@react-native-community/datetimepicker';
import {colors} from '../theme/colors'

interface dataPickerProps{
  datePickerText: string;
  defaultMode?: "date" | "time";
  isVisible?: boolean;
  date: Date;
  onDateChanges: (selectedDate: Date) => void;
}
export default function CustomDatePicker (props: dataPickerProps ) {
  const [mode, setMode] = useState<"date" | "time">( 'date');
  const [show, setShow] = useState<boolean>(false);
  const [localDate, setLocalDate] = useState<Date>(props.date);
  // pour quoi le event type n'est pas le mÃªme 
  const onChange = (event: any, selectedDate: Date | undefined) => {
    setShow(Platform.OS === 'ios');
   if (selectedDate){
    props.onDateChanges(selectedDate);
    setLocalDate(selectedDate);
   }
  };

  const showMode = (currentMode: "date" | "time") => {
    setShow(true);
    setMode(currentMode);
  };

  return (
    <View style={{ flexDirection:'row'}}>
      <Button onPress={()=>showMode('date')} title={props.datePickerText} />
      <Text style={{fontSize:16, color:colors.white}}> Date : {localDate.toDateString()} </Text>
      
      {show && (
        <DateTimePicker
          testID="dateTimePicker"
          value={props.date}
          mode={mode}
          is24Hour={true}
          display="default"
          onChange={onChange}
        />
      )}
    </View>
  );
};