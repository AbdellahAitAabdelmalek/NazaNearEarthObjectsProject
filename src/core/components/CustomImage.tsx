import React from 'react';
import {TouchableOpacity, Image, Alert} from 'react-native';

// comment specifier que imageUri est un string ?
export default function CustomImage({imageUri}:any) {

    return (
    <TouchableOpacity onPress={()=> Alert.alert("ImageClicked","You have clicked on the image",[ 
        {text: "Yes"},
        {text: "No"},
      ])
    }>
      {/* <Image style={styles.container} source={require("./assets/icon.png")} /> */}
      <Image 
          fadeDuration={1000}
          source={{
            width: 200,
            height:300,
            uri : imageUri,
          }}/>
    </TouchableOpacity>);
}