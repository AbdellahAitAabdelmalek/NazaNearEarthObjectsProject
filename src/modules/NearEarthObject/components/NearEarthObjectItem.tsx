import React, {useState} from 'react';
import {
    StyleSheet,
    Text,
    TouchableOpacity,
} from 'react-native';

  export default function NearEarthObjectItem({ item, onPress, style}:any) {
    return (
    <TouchableOpacity onPress={onPress} style={[styles.item, style]}>
        <Text style={styles.item}>{item.name}</Text>
      </TouchableOpacity>);
  }

  const styles = StyleSheet.create({
   item: {
      padding: 19,
      marginVertical: 8,
      marginHorizontal: 16,
    }
  });