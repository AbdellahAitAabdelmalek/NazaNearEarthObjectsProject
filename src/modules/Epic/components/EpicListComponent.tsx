import React, { FunctionComponent, useState, useContext } from "react";
import { StyleSheet, View } from "react-native";
import axios from "axios";
import { EpicPicture } from "../types/EpicPicture";
import LoadingData from "../../../core/components/LoadingData";
import { FlatList } from "react-native";
import { Dimensions } from "react-native";
import CustomImage from "../../../core/components/CustomImage";
import { StateContext } from "../../../pages/EpicReview/EpicReview.page";

export const EpicListComponent: FunctionComponent = () => {
  const states = useContext(StateContext);
  if (states === undefined) {
    throw new Error("CustomDatePicker must be used within a States Provider");
  }
  const [dataState, setDataState] = useState<"loading" | "available">(
    "loading"
  );
  const [listImage, setListImage] = useState<EpicPicture[]>();

  const localItem = ({ item }: any) => {
    const month: string = item.date.slice(5, 7);
    const day: string = item.date.slice(8, 10);
    const year: string = item.date.slice(0, 4);
    const imageUrl =
      "https://epic.gsfc.nasa.gov/archive/" +
      states.mode +
      "/" +
      year +
      "/" +
      month +
      "/" +
      day +
      "/png/" +
      item.image +
      ".png";
    const windowWidth = Dimensions.get("window").width * 0.9;
    return (
      <CustomImage
        width={windowWidth}
        height={windowWidth}
        imageUrl={imageUrl}
        resizeMode="center"
      />
    );
  };

  React.useEffect(() => {
    setDataState("loading");
    const month: string = ("0" + (states.date.getMonth() + 1)).slice(-2);
    const day: string = ("0" + states.date.getDate()).slice(-2);
    const year: string = "" + states.date.getFullYear();
    const url: string =
      "https://api.nasa.gov/EPIC/api/" +
      states.mode +
      "/" +
      year +
      "-" +
      month +
      "-" +
      day +
      "?api_key=y125lgm1Npphd8NEldDxfgTQ5q1NsnCsXzTgjqXw";
    axios
      .get(url)
      .then((res: any) => {
        const myList: EpicPicture[] = res.data.map(function (
          EpicPicture: EpicPicture
        ) {
          return {
            identifier: EpicPicture.identifier,
            caption: EpicPicture.caption,
            image: EpicPicture.image,
            version: EpicPicture.version,
            date: EpicPicture.date,
          };
        });
        setListImage(myList);
        setDataState("available");
      })
      .catch((_err: any) => {
        console.log("erreur !!!!");
      });
  }, [states]);

  return (
    <View style={styles.container}>
      {dataState === "loading" ? (
        <LoadingData loadingText="Image loading" />
      ) : (
        <FlatList
          style={styles.flatList}
          data={listImage}
          renderItem={localItem}
          keyExtractor={(item) => item.identifier.toString()}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  flatList: { width: "100%", flex: 1 },
});
