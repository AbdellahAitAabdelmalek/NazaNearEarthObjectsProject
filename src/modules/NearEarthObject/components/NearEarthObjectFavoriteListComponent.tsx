import React, { FunctionComponent } from "react";
import { NearEarthObject } from "../types/NearEarthObject.type";
import NearEarthObjectItem from "./NearEarthObjectItem";
import { connect } from "react-redux";

interface NearEarthObjectListProps {
  onItemIsPressed: (NearEarthObject: NearEarthObject) => void;
  dispatch: (action: Action) => void;
  favoriteObject: NearEarthObject[];
}
interface Action {
  type: string;
  value: NearEarthObject;
}

const NearEarthObjectListComponent: FunctionComponent<NearEarthObjectListProps> = (
  props: NearEarthObjectListProps
) => {
  return (
    <>
      {props.favoriteObject.map((nearEarthObject) => {
        return (
          <NearEarthObjectItem
            key={nearEarthObject.id}
            onItemIsPressed={props.onItemIsPressed}
            {...nearEarthObject}
          />
        );
      })}
    </>
  );
};

const mapStateToProps = (state: { favoriteObject: NearEarthObject[] }) => {
  return { favoriteObject: state.favoriteObject };
};

export default connect(mapStateToProps)(NearEarthObjectListComponent);
