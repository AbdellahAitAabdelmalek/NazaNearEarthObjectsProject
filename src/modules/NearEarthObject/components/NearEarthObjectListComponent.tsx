import React, { FunctionComponent,useState ,useEffect } from 'react';
import axios from 'axios';
import { NearEarthObjectItem } from './NearEarthObjectItem';
import { NearEarthObject } from '../types/NearEarthObject.type';
import LoadingData  from '../../../core/components/LoadingData';
import styled from 'styled-components/native';
import { FlatList } from 'react-native';

const url = 'http://www.neowsapp.com/rest/v1/neo/browse?page=0&size=20&api_key=y125lgm1Npphd8NEldDxfgTQ5q1NsnCsXzTgjqXw'

interface NearEarthObjectListProps {
    onItemIsPressed: (NearEarthObject: NearEarthObject) => void;
}

export const NearEarthObjectListComponent : FunctionComponent<NearEarthObjectListProps> = ({ onItemIsPressed }) => {
    
  const [selectedId, setSelectedId] = useState<number>(-1);
  const [listNearEarthObjects, setlistNearEarthObjects] = useState<NearEarthObject []>();
  const [isLoading, setisLoading] = useState <boolean>(true);
 
  const localItem = ({item}:any)=> {
    return (
      <NearEarthObjectItem 
        onItemIsPressed = {onItemIsPressed} 
        {...item}
      />);
    };

  React.useEffect(() => {
    axios
    .get(url)
    .then((res : any) =>{
      const myList:NearEarthObject [] = res.data.near_earth_objects.map( function (near_earth_object: NearEarthObject){ 
        return ({
          id: near_earth_object.id,
          neo_reference_id:near_earth_object.neo_reference_id,
          name:near_earth_object.name,
          name_limited: near_earth_object.name_limited,
          designation: near_earth_object.designation,
          nasa_jpl_url: near_earth_object.nasa_jpl_url,
          absolute_magnitude_h:near_earth_object.absolute_magnitude_h,
          is_potentially_hazardous_asteroid:near_earth_object.is_potentially_hazardous_asteroid,
          is_sentry_object: near_earth_object.is_sentry_object,
          });
      })
      setlistNearEarthObjects(myList);
      setisLoading(false);
    })
    .catch((_err : any)=> {
      console.log('erreur !!!!');
    })
  },[]);

  const FlatListCustom = styled.FlatList`
    width: 90%;
    background-color: white;
  `
  return ( 
      <>
        {isLoading === true ?
            <LoadingData/>
            :
            // pour quoi lorsque on utilise FlateList il y a plus de probl√®me de typage
            <FlatListCustom
              data={listNearEarthObjects}
              renderItem={localItem}
              keyExtractor={(item:any) => item.id.toString()}
            />
        }
      </>
    );
}