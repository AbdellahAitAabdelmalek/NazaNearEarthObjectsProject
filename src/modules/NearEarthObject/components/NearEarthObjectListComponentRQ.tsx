import React, { FunctionComponent, useState, useEffect } from "react";
import { Button, Text, View } from "react-native";
import { NearEarthObjectItem } from "./NearEarthObjectItem";
import { NearEarthObject } from "../types/NearEarthObject.type";
import styled from "styled-components/native";
import { usePaginatedQuery } from "react-query";
import { colors } from "../../../core/theme/colors";

const url =
  "http://www.neowsapp.com/rest/v1/neo/browse?api_key=y125lgm1Npphd8NEldDxfgTQ5q1NsnCsXzTgjqXw";

interface NearEarthObjectListProps {
  onItemIsPressed: (NearEarthObject: NearEarthObject) => void;
}

const fetchNearEarthObjects = async (key: string, page: number) => {
  const res = await fetch(url + "&page=" + page);
  return res.json();
};
export const NearEarthObjectListComponent: FunctionComponent<NearEarthObjectListProps> = ({
  onItemIsPressed,
}) => {
  const [listNearEarthObjects, setlistNearEarthObjects] = useState<
    NearEarthObject[]
  >([]);
  const [page, setPage] = useState<number>(0);
  const { resolvedData, latestData, status } = usePaginatedQuery(
    ["NearEarthObjects", page],
    fetchNearEarthObjects
  );

  const FlatListCustom = styled.FlatList`
    width: 90%;
    background-color: white;
  `;

  React.useEffect(() => {
    if (status === "success") {
      const myList: NearEarthObject[] = resolvedData.near_earth_objects.map(
        function (near_earth_object: NearEarthObject) {
          return {
            id: near_earth_object.id,
            neo_reference_id: near_earth_object.neo_reference_id,
            name: near_earth_object.name,
            name_limited: near_earth_object.name_limited,
            designation: near_earth_object.designation,
            nasa_jpl_url: near_earth_object.nasa_jpl_url,
            absolute_magnitude_h: near_earth_object.absolute_magnitude_h,
            is_potentially_hazardous_asteroid:
              near_earth_object.is_potentially_hazardous_asteroid,
            is_sentry_object: near_earth_object.is_sentry_object,
          };
        }
      );
      setlistNearEarthObjects(myList);
    }
  }, [status, page]);
  return (
    <>
      {status === "error" && (
        <Text style={{ color: colors.white }}> error </Text>
      )}
      {status === "loading" && (
        <Text style={{ color: colors.white }}> Loading ... </Text>
      )}
      {status === "success" && (
        <View
          style={{
            flex: 1,
            width: "100%",
            flexDirection: "column",
            justifyContent: "flex-end",
          }}
        >
          <View
            style={{
              flex: 1,
              flexDirection: "row",
              justifyContent: "space-around",
              alignItems: "flex-start",
            }}
          >
            <Button
              title="prev"
              onPress={() => setPage((old) => Math.max(old - 1, 1))}
            />
            <Text style={{ color: colors.white }}> {page} </Text>
            <Button
              title="next"
              onPress={() => setPage((old) => (!latestData ? old : old + 1))}
            />
          </View>
          <View
            style={{
              flex: 9,
              justifyContent: "flex-start",
              alignItems: "center",
            }}
          >
            <FlatListCustom
              data={listNearEarthObjects}
              renderItem={({ item }: any) => (
                <NearEarthObjectItem
                  onItemIsPressed={onItemIsPressed}
                  {...item}
                />
              )}
              keyExtractor={(item: any) => item.id.toString()}
            />
          </View>
        </View>
      )}
    </>
  );
};
