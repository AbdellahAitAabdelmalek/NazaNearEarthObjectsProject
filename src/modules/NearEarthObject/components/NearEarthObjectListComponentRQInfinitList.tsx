import React, { FunctionComponent, useState } from "react";
import {
  ActivityIndicator,
  FlatList,
  Text,
  View,
  StyleSheet,
  TextInput,
  Button,
} from "react-native";
import NearEarthObjectItem from "./NearEarthObjectItem";
import { NearEarthObject } from "../types/NearEarthObject.type";
import { colors } from "../../../core/theme/colors";

import { fetchNearEarthObjectList } from "../Hooks/NearEarthObjectListComponentRQInfinitListHooks";

interface NearEarthObjectListProps {
  onItemIsPressed: (NearEarthObject: NearEarthObject) => void;
}

export const NearEarthObjectListComponent: FunctionComponent<NearEarthObjectListProps> = ({
  onItemIsPressed,
}: NearEarthObjectListProps) => {
  const [listNearEarthObjects, setListNearEarthObjects] = useState<
    NearEarthObject[]
  >([]);
  const [
    listNearEarthObjectsFiltred,
    setListNearEarthObjectsFiltred,
  ] = useState<NearEarthObject[]>([]);
  const [searchedName, setsearchedName] = useState<string>("");
  const [page, setPage] = useState<number>(0);
  const {
    data,
    isLoading,
    isError,
    isSuccess,
    isPreviousData,
    isFetching,
  } = fetchNearEarthObjectList(page, searchedName);
  // filter listNearEarthObjects and put the result of that filter in ListNearEarthObjectsFiltred
  const _filterListObjectByName = () => {
    const myFiltredList = listNearEarthObjects.filter(
      (nearEarthObject: NearEarthObject) =>
        nearEarthObject.name.toLowerCase().includes(searchedName.toLowerCase())
    );
    setListNearEarthObjectsFiltred(myFiltredList);
  };

  React.useEffect(() => {
    console.log("states : " + isLoading, isError, isSuccess);
    console.log("Data : " + data);
    if (isSuccess) {
      const myFiltredList: NearEarthObject[] = data.near_earth_objects.map(
        function (near_earth_object: NearEarthObject) {
          return {
            id: near_earth_object.id,
            neo_reference_id: near_earth_object.neo_reference_id,
            name: near_earth_object.name,
            name_limited: near_earth_object.name_limited,
            designation: near_earth_object.designation,
            nasa_jpl_url: near_earth_object.nasa_jpl_url,
            absolute_magnitude_h: near_earth_object.absolute_magnitude_h,
            is_potentially_hazardous_asteroid:
              near_earth_object.is_potentially_hazardous_asteroid,
            is_sentry_object: near_earth_object.is_sentry_object,
          };
        }
      );
      setListNearEarthObjects([...listNearEarthObjects, ...myFiltredList]);
      // console.log("myFiltrerList : " + myFiltredList);
      // console.log("listNearEarthObjects" + listNearEarthObjects);
    }
  }, [data]);

  React.useEffect(() => {
    _filterListObjectByName();
  }, [searchedName, listNearEarthObjects]);
  return (
    <>
      <View
        style={{
          width: "90%",
          flex: 1,
          flexDirection: "row",
        }}
      >
        <TextInput
          placeholder='Near Earth Object Name'
          onChangeText={(text) => {
            setsearchedName(text);
          }}
          style={{ textAlign: "center", color: colors.black, flex: 1 }}
        />
        <Button title='Search' onPress={() => _filterListObjectByName()} />
      </View>

      <View
        style={{
          width: "90%",
          flex: 8,
          justifyContent: "flex-start",
          alignItems: "stretch",
        }}
      >
        {isSuccess && (
          <FlatList<NearEarthObject>
            data={listNearEarthObjectsFiltred}
            renderItem={({ item }: { item: NearEarthObject }) => (
              <NearEarthObjectItem
                onItemIsPressed={onItemIsPressed}
                {...item}
              />
            )}
            keyExtractor={(item: NearEarthObject) => item.id.toString()}
            onEndReachedThreshold={0.5}
            onEndReached={() => {
              console.log("onEndReached");
              setPage((old) => old + 1);
            }}
          />
        )}
        {isError && <Text style={{ color: colors.white }}> error </Text>}
        {isFetching && (
          <ActivityIndicator
            size='large'
            color='#00ff00'
            style={styles.activityIndicatorStyle}
          />
        )}
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  activityIndicatorStyle: {
    position: "absolute",
    left: 0,
    right: 0,
    top: 100,
    bottom: 0,
    alignItems: "center",
    justifyContent: "center",
  },
});
