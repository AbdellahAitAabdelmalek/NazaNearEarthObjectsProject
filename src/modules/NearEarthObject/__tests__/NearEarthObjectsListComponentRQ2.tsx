import { render, act } from "@testing-library/react-native";
import React from "react";
import { NearEarthObjectListComponent } from "../components/NearEarthObjectListComponentRQ";
import { NearEarthObject } from "../types/NearEarthObject.type";
import { QueryClient, QueryClientProvider } from "react-query";
import nock from "nock";

function tick() {
  return act(
    () =>
      new Promise((resolve) =>
        setTimeout(() => {
          resolve(0);
        }, 0)
      )
  );
}
const queryClient = new QueryClient();
const scope = nock("http://www.neowsapp.com")
  .get(
    "/rest/v1/neo/browse?api_key=y125lgm1Npphd8NEldDxfgTQ5q1NsnCsXzTgjqXw&page=0"
  )
  .reply(200, {
    near_earth_objects: [
      {
        id: 12,
        neo_reference_id: 13,
        name: "Abdel",
        name_limited: "name_limited",
        designation: "designation",
        nasa_jpl_url: "nasa_jpl_url",
        absolute_magnitude_h: 12,
        is_potentially_hazardous_asteroid: true,
        is_sentry_object: false,
      },
      {
        id: 13,
        neo_reference_id: 19,
        name: "Djamal",
        name_limited: "name_limited2",
        designation: "designation2",
        nasa_jpl_url: "nasa_jpl_url2",
        absolute_magnitude_h: 32,
        is_potentially_hazardous_asteroid: true,
        is_sentry_object: false,
      },
    ],
  });
scope.activeMocks();

test.only("should verify snapshot", async () => {
  const myNearEarthObjectListComponent = render(
    <QueryClientProvider client={queryClient}>
      <NearEarthObjectListComponent
        onItemIsPressed={(nearEarthObject: NearEarthObject) =>
          console.log("pressend" + nearEarthObject.id)
        }
      />
    </QueryClientProvider>
  );

  await tick();
  if (!scope.isDone()) {
    console.log("nock not intercepted");
  }

  expect(myNearEarthObjectListComponent).toMatchSnapshot();
});
