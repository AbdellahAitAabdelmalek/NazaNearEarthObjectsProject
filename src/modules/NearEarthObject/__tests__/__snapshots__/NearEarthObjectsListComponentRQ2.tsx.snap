// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should verify snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "90%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Previous Page
        </Text>
      </View>
    </View>
    <ActivityIndicator
      animating={true}
      color="#00ff00"
      hidesWhenStopped={true}
      size="large"
    />
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Next Page
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "stretch",
        "justifyContent": "flex-start",
        "width": "90%",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#00ff00"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 100,
        }
      }
    />
  </View>,
]
`;
